src/app/services/theme/theme.service.ts

```diff
@@ -41,109 +42,103 @@ export class ThemeService {
   /** Stored styles of the website */
   private style: CSSStyleDeclaration;
 
-  public constructor(@Inject(DOCUMENT) private document: Document) {
+  public constructor(
+    @Inject(DOCUMENT) private document: Document,
+    @Inject(IS_SERVER_PLATFORM) private isServer: boolean
+  ) {
     // Read the root style of the website
     this.style = this.document.documentElement.style;
   }
 
   /**
    * Set the base colour for a colour
    *
-   * @param color Colour to modify
-   * @param value New base value for colour (accepts most colour standards)
+   * @param colorName Colour to modify
+   * @param color New base value for colour (accepts most colour standards)
    */
-  public setColor(color: ThemeColor, value: string): void {
-    const hslValue = hsl(value);
-    if (
-      !hslValue ||
-      isNaN(hslValue.h) ||
-      isNaN(hslValue.s) ||
-      isNaN(hslValue.l)
-    ) {
-      console.log(hslValue);
-      console.warn(`Invalid theme color given for ${color} detected: ${value}`);
+  public setColor(colorName: ThemeColor, color: string): void {
+    const hslValue = hsl(color);
+
+    if (isNaN(hslValue.opacity)) {
+      console.warn(
+        `Invalid theme color given for ${colorName} detected: ${color}`
+      );
       return;
     }
-    this.setCssColorProperty(color, hslValue);
+
+    this.setCssColorProperty(colorName, hslValue);
   }
 
   /**
    * Set the global theme of the website
    *
    * @param theme Theme of website
    */
-  public setTheme(theme: BawTheme): void {
+  public setTheme(theme: BawTheme = {}): void {
     this.theme = theme;
-    this.customizeInstance();
+
+    // Re-add instance changes to theme
+    for (const themeColor of Object.keys(this.theme)) {
+      this.setColor(themeColor as ThemeColor, this.theme[themeColor]);
+    }
   }
 
   /**
    * Reset any modifications to a colour
    *
-   * @param color Color to reset
-   * @param hardReset Reset instance theme colour settings as well
+   * @param colorName Color to reset
    */
-  public resetColor(color: ThemeColor, hardReset?: boolean): void {
-    this.setCssColorProperty(color, null);
+  public resetColor(colorName: ThemeColor): void {
+    this.setCssColorProperty(colorName, null);
 
-    if (!hardReset) {
-      // Re-add instance changes to color
-      this.customizeInstance(color);
-    }
+    // Re-add instance changes to color
+    this.setColor(colorName, this.theme[colorName]);
```

Comment:
`setColor` itself calls `setCssProperty`, is there any reason for the additional `setColorProperty` above?